src/app/store/effects/download.effects.ts

```diff
@@ -66,13 +66,16 @@ export class DownloadEffects {
 
   private downloadNodes(toDownload: Array<MinimalNodeEntity>) {
     const nodes = toDownload.map(node => {
-      const { id, nodeId, name, isFile, isFolder } = <any>node.entry;
+      const { id, nodeId, name, isFile, isFolder, sharedByUser } = <any>(
+        node.entry
+      );
 
       return {
-        id: nodeId || id,
+        id: !!sharedByUser ? id : nodeId || id,
```

Comment:
```suggestion
        id: (toDownload.length === 1) ? (!!sharedByUser ? id : nodeId || id) : (nodeId || id),
```
this would fix the issue with download zip of shared files